Create 2022-04-01.MrMIME.md (#735)
mirage/mirage-www: data/2022-04-01.MrMIME.md

---
updated: 2022-04-01
authors:
- name: Romain Calascibetta
  uri: https://github.com/dinosaure
subject: 'Email in OCaml'
permalink: 2022-04-01_MrMIME
---

# Email in OCaml

The security of communications poses a seemingly never-ending challenge across Cyberspace. From sorting through mountains of spam to protecting our private messages from malicious hackers, cybersecurity has never been more important than it is today. It takes considerable technical skills and dependable infrastructure to run an email service, and sadly, most companies with the ability to handle the billions of emails sent daily make money off mining your sensitive data. 

> Five years ago, we started to explore an incredible endeavour on how to securely send and receive email. It was my final year in an internship at Cambridge, and the goal was to develop an OCaml library that could _parse_ and _craft_ emails. Thus, Mr. MIME was born. I even gave a [presentation on it at ICFP 2016](https://www.youtube.com/watch?v=kQkRsNEo25k) and introduced Mr MIME in a [previous post](https://tarides.com/blog/2019-09-25-mr-mime-parse-and-generate-emails).

Reviewer: ```suggestion
Five years ago, we started to explore an incredible endeavour on how to securely send and receive email. It was my final year in an internship at Cambridge, and the goal was to develop an OCaml library that could _parse_ and _craft_ emails. Thus, Mr. MIME was born. I even gave a [presentation on it at ICFP 2016](https://www.youtube.com/watch?v=kQkRsNEo25k) and introduced Mr MIME in a [previous post](https://tarides.com/blog/2019-09-25-mr-mime-parse-and-generate-emails). Mr MIME also has been selected by the [NGI DAPSI initiative]((https://tarides.com/blog/2022-03-08-secure-virtual-messages-in-a-bottle-with-scop) last year.
```

> Especially since we've passed both [Phase 1](https://tarides.com/blog/2021-10-14-scop-selected-for-dapsi-phase2) and [Phase 2](https://tarides.com/blog/2022-03-08-secure-virtual-messages-in-a-bottle-with-scop) of the [the DAPSI initiative](https://dapsi.ngi.eu/), I'm thrilled to shine a spotlight on Mr. MIME in conjunction with the [MirageOS 4 release](https://mirage.io/blog/announcing-mirage-40)! It was essential to create several small libraries when building and testing Mr. MIME. I've included some samples of how to use Mr. MIME to parse and serialise emails in OCaml, as well as receiving and sending SMTP messages. I then explain how to use all of this via CLI tools. Since *unikernels* were the foundation on which I built Mr. MIME, the final section explains how to deploy unikernels to handle email traffic. 

Reviewer: ```suggestion
I'm thrilled to shine a spotlight on Mr. MIME as part of the [MirageOS 4 release](https://mirage.io/blog/announcing-mirage-40)! It was essential to create several small libraries when building and testing Mr. MIME. I've included some samples of how to use Mr. MIME to parse and serialise emails in OCaml, as well as receiving and sending SMTP messages. I then explain how to use all of this via CLI tools. Since *unikernels* were the foundation on which I built Mr. MIME, the final section explains how to deploy unikernels to handle email traffic. 
```

> Five years ago, we started to explore an incredible endeavour on how to securely send and receive email. It was my final year in an internship at Cambridge, and the goal was to develop an OCaml library that could _parse_ and _craft_ emails. Thus, Mr. MIME was born. I even gave a [presentation on it at ICFP 2016](https://www.youtube.com/watch?v=kQkRsNEo25k) and introduced Mr MIME in a [previous post](https://tarides.com/blog/2019-09-25-mr-mime-parse-and-generate-emails). Mr MIME also has been selected by the [NGI DAPSI initiative]((https://tarides.com/blog/2022-03-08-secure-virtual-messages-in-a-bottle-with-scop) last year.

Author: ```suggestion
Five years ago, we started to explore an incredible endeavour on how to securely send and receive email. It was my final year in an internship at Cambridge, and the goal was to develop an OCaml library that could _parse_ and _craft_ emails. Thus, Mr. MIME was born. I even gave a [presentation on it at ICFP 2016](https://www.youtube.com/watch?v=kQkRsNEo25k) and introduced Mr. MIME in a [previous post](https://tarides.com/blog/2019-09-25-mr-mime-parse-and-generate-emails). Mr. MIME was also selected by the [NGI DAPSI initiative]((https://tarides.com/blog/2022-03-08-secure-virtual-messages-in-a-bottle-with-scop) last year.
```

Author: I'm not sure what happened with this commit (https://github.com/mirage/mirage-www/pull/735/commits/b662fc1edc86a251820debc40057f407194fab77), so I hope I didn't overwrite this file! Please let me know if I did, and I'll put it back up.
